classes/Language.php

```diff
@@ -1342,75 +1342,68 @@ public static function getLanguagePackListContent($iso, $tar)
-                }
-            }
-
-            $gz = new Archive_Tar($filegz, true);
-            if (!$gz) {
-                continue;
-            }
-            $files_list = Language::getLanguagePackListContent($lang['iso_code'], $gz);
-            foreach ($modules_list as $module_name) {
-                foreach ($files_list as $i => $file) {
-                    if (strpos($file['filename'], 'modules/' . $module_name . '/') !== 0) {
-                        unset($files_list[$i]);
-                    }
-                }
-            }
-            foreach ($files_list as $file) {
-                if (isset($file['filename']) && is_string($file['filename'])) {
-                    $files_listing[] = $file['filename'];
-                }
-            }
-            $gz->extractList($files_listing, _PS_TRANSLATIONS_DIR_ . '../', '');
+            static::updateMultilangTable($lang['iso_code']);
         }
     }
 
     /**
      * Update all table_lang from xlf & DataLang.
      *
-     * @param $iso_code
+     * @param string $iso_code 2-letter language code
      *
      * @return bool
      */
     public static function updateMultilangTable($iso_code)
     {
-        $useLang = Db::getInstance()->getRow('SELECT * FROM `' . _DB_PREFIX_ . 'lang` WHERE `iso_code` = "' . pSQL($iso_code) . '" ', true, false);
+        $langId = Language::getIdByIso($iso_code);
 
-        if (!empty($useLang)) {
-            $lang = new Language($useLang['id_lang']);
+        if (!empty($langId)) {
+            $lang = new Language($langId);
 
-            $tables = Db::getInstance()->executeS('SHOW TABLES LIKE \'' . str_replace('_', '\\_', _DB_PREFIX_) . '%\_lang\' ');
-            foreach ($tables as $table) {
-                foreach ($table as $t) {
-                    $className = ucfirst(Tools::toCamelCase(str_replace(_DB_PREFIX_, '', $t)));
-
-                    if (_DB_PREFIX_ . 'country_lang' == $t) {
-                        self::updateMultilangFromCldr($lang);
-                    } else {
-                        self::updateMultilangFromClass($t, $className, $lang);
-                    }
-                }
+            $rows = Db::getInstance()->executeS('SHOW TABLES LIKE \'' . str_replace('_', '\\_', _DB_PREFIX_) . '%\_lang\' ');
+            if (!empty($rows)) {
+                // get all values
+                $tableNames = array_map('reset', $rows);
+                static::updateMultilangTables($lang, $tableNames);
             }
 
-            Hook::exec('actionUpdateLangAfter', ['lang' => $lang]);
         }
 
         return true;
     }
 
+    /**
+     * Translates translatable content in the requested database tables
+     *
+     * @param Language $language Language to translate to
+     * @param string[] $tablesToUpdate Tables to update (including datbase prefix, ending in _lang)
+     *
+     * @throws PrestaShopException
+     */
+    public function updateMultilangTables(Language $language, array $tablesToUpdate)
```

Comment:
Change to static please ! See @https://github.com/eternoendless/PrestaShop/blob/fix-things/classes/Language.php#L1376